### PR TITLE
Remove Deprecated Code Coverage Documentation in Pester

### PR DESCRIPTION
**Pull request by mahomedalid**
Description: This pull request (PR) aims to enhance the Pester documentation by removing outdated information related to code coverage configuration. The documentation currently includes details about using the -CodeCoverage parameter with Invoke-Pester, which has been deprecated in favor of the more modern New-PesterConfiguration method. This PR updates the documentation to provide up-to-date instructions on configuring and generating code coverage metrics using New-PesterConfiguration. 

By doing so, it ensures that users receive accurate and relevant information when working with Pester's code coverage feature, thus improving the overall usability of the documentation (and avoid frustration like the one I had myself!)

---
Solves #196 
@fflaten  as we discussed this is an updated version of Code Coverage. Let me know what you think!

### CODE REVIEW

File: docs/usage/code-coverage.mdx [hunk 1]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
+# deploy-dev-smple.ps1                                                                                            18 Write-Ou...
+# deploy-dev-sample.ps1                                                                                           20 Write-Er...
+# deploy-dev-sample.ps1                                                                                           22 $Content
+# deploy-dev-sample.ps1                                                                                           22 Set-Cont...
+# deploy-dev-sample.ps1                                                                                           24 kubectl ...
+# jade-codes.dev.ps1                                                                                              1  Show-Out...
 ```
 
-## Publishing Code Coverage Metrics In An [Azure DevOps](https://azure.microsoft.com/en-us/solutions/devops/) Pipeline
+## Understanding Code Coverage
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:81]: Duplicate content. See first section of page.

File: docs/usage/code-coverage.mdx [hunk 2]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
-#>
+```powershell
+$config = New-PesterConfiguration
+$config.CodeCoverage.Enabled = $true
+$config.Output.Verbosity = 'Detailed'
+
+Invoke-Pester -Configuration $config
+
+# ...
+# deploy-dev-smple.ps1                                                                                            18 Write-Ou...
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:73]: Please format this or generate new output using a smaller console window. Too much whitespace between filename and function.

File: docs/usage/code-coverage.mdx [hunk 3]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
+Invoke-Pester -Configuration $config
+```
 
-<#
-Code coverage report:
-Covered 66.67 % of 3 analyzed commands in 1 file.
+For additional `CodeCoverage` configuration options, please refer to the [New-PesterConfiguration](https://pester.dev/docs/commands/New-PesterConfiguration) documentation.
 
-Missed commands:
+> **Note:** Pester does not traverse directories.
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:54]: Maybe this is better as a tip admonition? Example  https://github.com/pester/docs/blob/e30ea3b3c5cc78507965d952e744c1f926482493/docs/usage/configuration.mdx#L19-L21

File: docs/usage/code-coverage.mdx [hunk 4]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
-<#
-Code coverage report:
-Covered 66.67 % of 3 analyzed commands in 1 file.
+For additional `CodeCoverage` configuration options, please refer to the [New-PesterConfiguration](https://pester.dev/docs/commands/New-PesterConfiguration) documentation.
 
-Missed commands:
+> **Note:** Pester does not traverse directories. 
+If you encounter an **empty `coverage.xml` file**, it's likely because you're running the tests from a directory that doesn't
+include the relevant code to be covered. 
+To resolve this, you can either follow the recommended [Test file structure](https://pester.dev/docs/usage/test-file-structure),
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:57]: There's a `CodeCoverage.Path` that should be prefered. A parent folder may include other files like build/release/test scripts etc. that you don't want to include.

File: docs/usage/code-coverage.mdx [hunk 5]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
-- `Function` or `f`: The name of a function you wish to analyze within a file. This value may contain wildcards, and any matching functions will be analyzed. If this key is used, `StartLine` and `EndLine` are ignored for this particular hashtable.
-- `StartLine` or `s`: The first line of a range to be analyzed. If this key is used and no corresponding value is assigned to `EndLine`, then the entire remainder of the file starting with `StartLine` will be analyzed.
-- `EndLine` or `e`: The last line of a range to be analyzed. If this key is used and no corresponding value is assigned to `StartLine`, the entire file up to and including `EndLine` will be analyzed.
-
-After `Invoke-Pester` finishes executing the test scripts, Pester will output a coverage report to the console. If you are using `Invoke-Pester`'s `-PassThru` switch, the coverage analysis will also be available on the output object, under its `CodeCoverage` property.
-
-Here are some examples of the various ways the `-CodeCoverage` parameter can be used, and their corresponding output when used with the following files.
-
-```powershell title="CoverageTest.ps1"
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:28]: Keep the example files + matching output for the inital demo? Maybe easier to undestand the concept for new users?

File: docs/usage/code-coverage.mdx [hunk 6]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
-function FunctionTwo
-{
-    return 'I get executed'
-    return 'I do not'
-}
-```
+## Code Coverage Quickstart with New-PesterConfiguration
 
-```powershell title="CoverageTest.Tests.ps1"
-$here = Split-Path -Parent $MyInvocation.MyCommand.Path
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:49]: If samples are reused: Needs to be updated to v5 syntax with `BeforeAll` and `$PSScriptRoot`

File: docs/usage/code-coverage.mdx [hunk 7]
```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
...
-    }
-}
-
-function FunctionTwo
-{
-    return 'I get executed'
-    return 'I do not'
-}
-```
+## Code Coverage Quickstart with New-PesterConfiguration
```
Comment (fflaten) [inline docs/usage/code-coverage.mdx:10]:
```suggestion
## Quickstart
```

Shorten title?

**Review by fflaten — CHANGES_REQUESTED** [review]
Great work! I've added a few comments from what I could review while on travel. 🙁

File: docs/usage/code-coverage.mdx (hunk 4 above)
Comment (mahomedalid) [reply docs/usage/code-coverage.mdx:64]: This is cool, thanks for the suggestion.

File: docs/usage/code-coverage.mdx (hunk 6 above)
Comment (mahomedalid) [reply docs/usage/code-coverage.mdx:49]: Updated.

File: docs/usage/code-coverage.mdx (hunk 5 above)
Comment (mahomedalid) [reply docs/usage/code-coverage.mdx:28]: Old example files integrated.

**Comment by mahomedalid** [comment]
@fflaten solved all of them, safe travels.

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
LGTM. Thanks :)

**Comment by fflaten** [comment]
LGTM. This was long overdue, so really appreciate the work and time you've spent here @mahomedalid 👏

**Pull request merged**